lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -82,4 +90,11 @@ defmodule Mix.Tasks.Escript.Install do
                   <> "to be able to invoke escripts by name."
     end
   end
+
+  defp is_escript?(binary) do
+    case binary_part(binary, 0, 2) do
+      "#!" -> :ok
+      _ -> :undefined
+    end
+  end
```

Comment:
Not sure if there's a particular reason why you did `:ok | :undefined`, but we usually have this name convention in Elixir:

* `is_x` is reserved for predicates (thus returning a `boolean :: true | false`) that can be used in guards
* `x?` should return a boolean

I think in this case it's safe to have:

```elixir
defp escript?(binary) do
  binary_part(binary, 0, 2) == "#!"
end
```

and change the inner `case` inside `before_install/2` to be an `if`. Wdyt?